locobot/agent/perception/handlers/core.py

```diff
@@ -73,26 +73,28 @@ class RGBDepth:
     Args:
         rgb (np.array): RGB image fetched from the robot
         depth (np.array): depth map fetched from the robot
-        pts (list[(x,y,z)]): list of x,y,z coordinates of the pointcloud corresponding 
+        pts (np.array [(x,y,z)]): array of x,y,z coordinates of the pointcloud corresponding 
         to the rgb and depth maps.
     """
     rgb: np.array
     depth: np.array
-    ptcloud: list
+    ptcloud: np.array
 
     def __init__(self, rgb, depth, pts):
         self.rgb = rgb
         self.depth = depth
-        self.ptcloud = pts
+        self.ptcloud = pts.reshape(rgb.shape)
```

Comment:
are the points in the point cloud laid out in the same co-ordinates as the RGB image? i.e. you should confirm visually that `pts.reshape(rgb.shape)` does what you expect

Reply:
Yes. `pts` previously was a (w*h, 3) shaped array (where h and w are height and width of the image respectively) created in row-major form. I wanted to convert it to a (w,h,3) shaped array (which is the shape of the rgb image), so that we have 3 channels (x,y,z) for each pixel in the image. 

Since pts originally is saved in row-major form (or C-like indexing where the last index changes fastest), reshaping works since it uses C-like indexing to reshape the array by default (more [here](https://numpy.org/doc/stable/reference/generated/numpy.reshape.html)). 

Tested this with the following code snippet, and some sanity checking in habitat to gather enough confidence to merge. But it also needs testing on the robot, which I'll do later this week.

![Screen Shot 2021-05-05 at 12 18 32 PM](https://user-images.githubusercontent.com/57542204/117174771-11cbbb80-ad9c-11eb-9ace-03ce880ebb93.png)